src/ui/public/vislib/lib/axis.js

```diff
@@ -15,15 +15,17 @@ export default function AxisFactory(Private) {
   const AxisConfig = Private(AxisConfigProvider);
 
   class Axis extends ErrorHandler {
-    constructor(args) {
+    constructor(config, args) {
       super();
-      this.data = args.data;
-      this.config = new AxisConfig(args);
+      this.data = config.data;
```

Comment:
Why not make the chart config and data separate things? Only benefit that's immediately obvious to me that it's one variable instead of two, but I would argue that actually makes the dependencies of `Axis` less obvious.